test/core/unwind.wast

```diff
@@ -70,6 +70,11 @@
       (i32.const 3) (block (i64.const 1) (return (i32.const 9)))
     )
   )
+  (func (export "br_if-unwind-by-last-value") (param i32) (result i32)
```

Comment:
Do you still plan to address this comment as well?

Reply:
@rossberg, thanks for your review! I am OK to address it in this PR.

Reply:
@rossberg, I fail to compile it when moving the unused operand before `br_if`, see code snippet below:
```
(func (export "block-unwind-by-br_if-value") (result i32)
  (block (result i32)
    (i32.const 3) (i64.const 1) (br_if 0 (i32.const 9) (i32.const 1))
  )
)
```

Is it a bug or by design?

Reply:
This doesn't have anything to do with the syntax. But because br_if is not an unconditional branch, you have to insert a couple of `drops` after it to flush the superflous stack operands on the path where the branch is not taken. (In your previous version, the `return` had the same effect.)

Reply:
Many thanks for your explanation! Really useful!
I just add a new commit, but some tests are still failed to be compiled. Their root cause look like the same, the operator type before `br_if` and the result type are mismatched, I suppose that's why it fails but have no idea on how to fix it. :(

I commented out these tests, Could you PTAL?